x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/api/review_rule_upgrade/route.ts

```diff
@@ -114,28 +115,27 @@ const getRuleDiffCalculationArgs = (
     const baseRule = baseRulesMap.get(ruleId);
     const latestRule = latestRulesMap.get(ruleId);
 
-    // TODO: https://github.com/elastic/kibana/issues/148189
-    // Make base versions optional for diff calculation. We need to support this in order to be able
-    // to still show diffs for rule assets coming from packages without historical versions.
-    if (installedRule != null && baseRule != null && latestRule != null) {
-      result.push({
-        currentVersion: installedRule,
-        baseVersion: baseRule,
-        targetVersion: latestRule,
-      });
-    }
+    // baseRule can be undefined if the rule has no historical versions, but other versions should always be present
+    invariant(installedRule != null, `installedRule is not found for rule_id: ${ruleId}`);
+    invariant(latestRule != null, `latestRule is not found for rule_id: ${ruleId}`);
+
+    result.push({
+      currentVersion: installedRule,
+      baseVersion: baseRule,
+      targetVersion: latestRule,
+    });
   });
 
   return result;
 };
 
 const calculateRuleStats = (results: CalculateRuleDiffResult[]): RuleUpgradeStatsForReview => {
+  const allTags = new Set<string>(
+    results.flatMap((result) => result.ruleVersions.input.current.tags)
+  );
```

Comment:
We're not doing it for the [`installation/_review` route](https://github.com/elastic/kibana/blob/03cdec3afcfb4a0f54db5e51dbf059c22f755de5/x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/api/review_rule_installation/route.ts#L80-L83) either, but perhaps we could do a default alphabetical sort on the tags? 

The tags component handles this now, but it's a nice default for future consumers of this API (though this is internal, so no biggie either way).

Reply:
I think it's a good idea 👍